tokio/src/io/util/lines.rs

```diff
@@ -59,6 +59,21 @@ where
 
         poll_fn(|cx| Pin::new(&mut *self).poll_next_line(cx)).await
     }
+
+    /// Obtain a mutable reference to a byte of the underlying reader
+    pub fn get_mut(&mut self) -> &mut R {
+        &mut self.reader
+    }
+
+    /// Obtain a reference to a byte of the underlying reader
+    pub fn get_ref(&mut self) -> &R {
+        &self.reader
+    }
+
+    /// Consume the line, returning the underlying bytes
```

Comment:
The docs in `std` for a comparable method ([`BufReader::into_inner`](https://doc.rust-lang.org/stable/std/io/struct.BufReader.html#method.into_inner)) say:
> Unwraps this `BufReader<R>`, returning the underlying reader.
so maybe this should be 
```suggestion
    /// Unwraps this `Lines`, returning the underlying reader.
```

Also, `std`'s docs warn that
> Note that any leftover data in the internal buffer is lost. Therefore, a following read from the underlying reader may lead to data loss.

This seems to be the case here, as well: if `into_inner` is called while part of an unfinished line is buffered, that line will be lost. Our documentation should probably also warn about this.

Reply:
Resolved with @7dc85efbf3f9d402682ab007f589d87316938bec